modules/programs/git.nix

```
@@ -479,6 +514,45 @@ in {
         };
     })
 
+    (mkIf cfg.maintenance.enable {
+      programs.git.iniContent.maintenance.repo = cfg.maintenance.repositories;
+
+      systemd.user.services."git-maintenance@" = {
+        Unit.Description = "Optimize Git repositories data";
+
+        Service = {
+          Type = "oneshot";
+          ExecStart = let exe = lib.getExe cfg.package;
+          in ''
+            "${exe}" --exec-path="${exe}" for-each-repo --config=maintenance.repo maintenance run --schedule=%i
+          '';
+          LockPersonality = "yes";
+          MemoryDenyWriteExecute = "yes";
+          NoNewPrivileges = "yes";
+          RestrictAddressFamilies = "AF_UNIX AF_INET AF_INET6 AF_VSOCK";
+          RestrictNamespaces = "yes";
+          RestrictRealtime = "yes";
+          RestrictSUIDSGID = "yes";
+          SystemCallArchitectures = "native";
+          SystemCallFilter = "@system-service";
+        };
+      };
+
+      systemd.user.timers = let
+        toSystemdTimer = name: time:
+          lib.attrsets.nameValuePair "git-maintenance@${name}" {
+            Unit.Description = "Optimize Git repositories data";
+
+            Timer = {
+              OnCalendar = time;
+              Persistent = true;
+            };
+
+            Install.WantedBy = [ "timers.target" ];
+          };
+      in lib.attrsets.mapAttrs' toSystemdTimer cfg.maintenance.timers;
+    })
```

Maybe you could consider templating the timer as well? It doesn't carry a lot of benefit, but it avoids copy pasting logic and lets users start ad-hoc timers with an `OnCalendar` field of their choosing. We could also potentially drop the `OnCalendar` overrides and let the template handle it, since it takes `daily`, `weekly`, etc. just fine on it's own.

```suggestion
      systemd.user.timers = {
        "git-maintenance@" = {
          Unit.Description = "Optimize Git repositories data";
          
          Timer = {
            OnCalendar = "%i";
            Persistent = true;
          };
          
          Install.WantedBy = [ "timers.target" ];
        };
      }
      // 
      lib.attrsets.mapAttrs'
        (n: v: lib.attrsets.nameValuePair "git-maintenance@${n}" {
          Timer.OnCalendar = v;
        })
        cfg.maintenance.timers;
```

---

Good idea! Done [here](https://github.com/nix-community/home-manager/compare/39432cd042226d331207ccf22392264f7f0ed3c7..bcf8990114a66bfd992c4dce8da2c6d6ed8ae5f3)

---

Looks like this produces an error `Unit name git-maintenance@.timer is missing the instance name.` because `sd-switch` tries to start `git-maintenance@.timer` and not `git-maintenance@something.timer`. Reported this to `sd-switch`'s [bug tracker](https://lists.sr.ht/~rycee/public-inbox/%3CCADPR+QweDcpom2h-6xv1+zRDjybcv7KMTagqQ-CBZJJrMEx7=A@mail.gmail.com%3E)